.github/workflows/build_and_test_dbt.yaml

```diff
@@ -57,6 +57,40 @@ jobs:
         working-directory: ${{ env.PROJECT_DIR }}
         shell: bash
 
+      - name: Deploy model dependencies
+        run: |
+          if [[ $MODIFIED_RESOURCES_ONLY == 'true' ]]; then
+            if [[ $MANUALLY_DISPATCHED == 'true' ]]; then
+              echo "Deploying dependencies for manually selected resources"
+              ../.github/scripts/deploy_dbt_model_dependencies.sh ${{ inputs.models }}
+            else
+              echo "Deploying dependencies for modified/new resources only"
+              model_names=$(\
+                dbt list \
+                  --quiet -t "$TARGET" -s state:modified state:new \
+                  --state "$STATE_DIR" --resource-types model \
+                  --output json --output-keys name | \
+                jq -r '.name' | \
+                tr '\n' ' '
+              )
+              if [ -z "$model_names" ]; then
+                echo "No modified/new resources found, skipping dependency deploy"
```

Comment:
See [this workflow run](https://github.com/ccao-data/data-architecture/actions/runs/9024036185/job/24797106700?pr=435#step:7:53) for an example of this conditional branch in action.